tests/callbacks/test_TimerListener.cpp

```diff
@@ -0,0 +1,109 @@
+ */
+
+using execution_space = Kokkos::DefaultExecutionSpace;
+
+namespace Kokkos::utils::tests::callbacks
+{
+
+using namespace Kokkos::utils::callbacks;
+
+//! Listener to time parallel for regions.
+using parallel_for_timer_t = ParallelForTimerListener<EventRegexMatcher, execution_space>;
+
+class TimerListenerTest : public ManagerTestFixture
+{
+public:
+    void SetUp() override {
+        this->exec = Kokkos::Experimental::partition_space(execution_space{}, 1)[0];
+    }
+
+protected:
+    execution_space exec {};
+};
+
+//! @test Check traits of @ref Kokkos::utils::callbacks::ParallelForTimerListener.
+TEST(ParallelForTimerListener, traits)
+{
+    /// Check that @ref parallel_for_timer_t is a listener for @ref Kokkos::utils::callbacks::BeginParallelForEvent
+    /// and @ref Kokkos::utils::callbacks::EndParallelForEvent.
+    static_assert(std::same_as<
+        listener_event_type_list_t<parallel_for_timer_t>,
+        Kokkos::Impl::type_list<BeginParallelForEvent, EndParallelForEvent>
+    >);
+
+    static_assert(Listener<parallel_for_timer_t>);
+}
+
+//! @test Check the behavior of @c Kokkos::utils::callbacks::ParallelForTimerListener.
+TEST_F(TimerListenerTest, parallel_for)
+{
+    const auto par_for_timer_matching = std::make_shared<parallel_for_timer_t>(exec, EventRegexMatcher(std::regex("computation - level 0 - pfor")));
+    const auto par_for_timer_no_match = std::make_shared<parallel_for_timer_t>(exec, EventRegexMatcher(std::regex("this-should-not-match")));
+
+    Manager::register_listener(par_for_timer_matching);
+    Manager::register_listener(par_for_timer_no_match);
+
+    MyWorkload<execution_space>{}.execute(exec);
+
+    ASSERT_GT(par_for_timer_matching->timer.duration(), Kokkos::utils::timer::seconds(0));
+
+    EXPECT_FALSE(par_for_timer_no_match->connected() || par_for_timer_no_match->closed())
+        << "The listener should not be connected or closed because the events did not match.";
+
+    Manager::unregister_listener(par_for_timer_matching.get());
+    Manager::unregister_listener(par_for_timer_no_match.get());
+}
+
+//! Listener to time profile regions.
+using region_timer_t = RegionTimerListener<EventRegexMatcher, execution_space>;
+
+//! @test Check traits of @c Kokkos::utils::callbacks::RegionTimerListener.
+TEST(RegionTimerListener, traits)
+{
+    /// Check that @ref region_timer_t is a listener for @ref Kokkos::utils::callbacks::PushRegionEvent
+    /// and @ref Kokkos::utils::callbacks::PopRegionEvent.
+    static_assert(std::same_as<
+        listener_event_type_list_t<region_timer_t>,
+        Kokkos::Impl::type_list<PushRegionEvent, PopRegionEvent>
+    >);
+
+    static_assert(Listener<region_timer_t>);
+}
+
+//! @test Check the behavior of @c Kokkos::utils::callbacks::RegionTimerListener.
+TEST_F(TimerListenerTest, region)
+{
+    const auto par_for_timer = std::make_shared<parallel_for_timer_t>(exec, EventRegexMatcher(std::regex("computation - level 0 - pfor")));
```

Comment:
Using my suggestion to loosen the constructor constraint ?
```suggestion
    const auto par_for_timer = std::make_shared<parallel_for_timer_t>(exec, std::regex("computation - level 0 - pfor"));
```

Reply:
Indeed. Still a todo for me.

Reply:
OK! Works now indeed! Good suggestion!